proposals/sdk-packaging-consolidation/README.md

```diff
@@ -0,0 +1,115 @@
+
+Consolidating these packages will simplify the user experience ("just install kfp") and streamline the development and release process.
+
+### Goals
+
+1.  Consolidate `kfp-pipeline-spec`, `kfp-server-api`, and `kfp-kubernetes` into the `kfp` package.
+2.  Migrate the build system from `setup.py` to `pyproject.toml` (PEP 621).
+3.  Ensure all existing functionality and tests are preserved.
+4.  Simplify the installation process for end-users.
+
+### Non-Goals
+
+1.  Major refactoring of the internal logic of these components (other than what's needed for consolidation).
+2.  Changing the public API surface significantly (backward compatibility should be maintained where possible, though import paths may change).
+
+## Proposal
+
+### Package Consolidation
+
+The following packages will be merged into `kfp`:
+
+*   **`kfp-pipeline-spec`**: Will move to `kfp.pipeline_spec`.
+*   **`kfp-server-api`**: Will move to `kfp.server_api`.
+*   **`kfp-kubernetes`**: Will move to `kfp.kubernetes`.
+
+### Build System Modernization
+
+We will move away from `setup.py` and `requirements.txt` files to a single `pyproject.toml` file. This aligns with modern Python packaging standards and simplifies dependency specification.
+
+### Version Alignment
+
+The consolidated package will use a unified version number (starting with 3.0.0) to signify this major structural change and ensure users get compatible versions of all components.
+
+## Design Details
+
+### Directory Structure
+
+The new structure within `sdk/python/kfp/` will be:
+
+```
+sdk/python/kfp/
+├── pipeline_spec/       # Formerly kfp-pipeline-spec
+├── server_api/          # Formerly kfp-server-api
+├── kubernetes/          # Formerly kfp-kubernetes
+├── dsl/
+├── client/
+└── ...
+```
+
+### Namespace Handling
+
+We will update internal imports to use relative imports or full `kfp.*` paths. For example, `import kfp_server_api` will become `from kfp import server_api` or `import kfp.server_api`.
+
+## Test Plan
+
+1.  **Unit Tests**: Migrate existing unit tests from the separate repositories/directories into the `kfp` test suite.
+    *   Specifically, `kubernetes_platform` tests will be moved to `sdk/python/kfp/kubernetes/test/`.
+2.  **Integration Tests**: Verify that the consolidated package works with existing integration tests.
+3.  **Installation Tests**: Verify that `pip install .` and `pip install kfp` work correctly in a fresh environment.
+
+## Migration Strategy
+
+Users will simply need to upgrade to the new `kfp` version:
+
+```bash
+pip install --upgrade kfp
+```
+
+This will install the consolidated package. The old packages (`kfp-pipeline-spec`, etc.) will no longer be needed as dependencies, though they may remain on PyPI for older versions.
+
+**Breaking Changes**:
+*   Users directly importing `kfp_pipeline_spec`, `kfp_server_api`, or `kfp_kubernetes` will need to update their imports to `kfp.pipeline_spec`, `kfp.server_api`, and `kfp.kubernetes` respectively.
```

Comment:
`kfp_pipeline_spec` and `kfp_kubernetes` are `kfp`-namespaced package already. The imports are already `kfp.pipeline_spec` and `kfp.kubernetes` there's no change how they are imported. Only change is how they're installed.